### PR TITLE
implement i128

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by yjhmelody** [comment]
Is there any problem with this PR ?

**Comment by MaxGraey** [comment]
@yjhmelody No, I just need to take the time to finish this.

File: assembly/integer/i128.ts [hunk 1]
```diff
@@ -90,28 +90,60 @@ export class i128 {
...
+  }
 
-    return new i128(lo, hi);
+  static fromUint8ArrayBE(array: Uint8Array): i128 {
+    assert(array.length && (array.length & 15) == 0);
+    // @ts-ignore
+    var buffer = array.dataStart;
+    return new i128(
+      bswap<u64>(load<u64>(buffer, 1 * sizeof<u64>())),
+      bswap<u64>(load<u64>(buffer, 0 * sizeof<u64>()))
```
Comment (yjhmelody) [inline assembly/integer/i128.ts:145]: not `bswap<i64>(load<i64>(buffer, 0 * sizeof<i64>()))`?

Comment (MaxGraey) [reply assembly/integer/i128.ts:145]: it doesn't matter in this case

**Pull request merged**